### ISSUE TITLE
[Feature] omit setting gt labels to be zeros in RPN Head

### ISSUE THREAD

**Issue by GeorgePearse** [BinItAI/visdet#803]
**Original issue:** https://github.com/open-mmlab/mmdetection/issues/9182
**Created:** 2022-10-30
**Last updated:** 2022-10-31

---

### What's the feature?

In RPN head, there are some legacy codes do the following:
```python
# set cat_id of gt_labels to 0 in RPN
            for data_sample in rpn_data_samples:
                data_sample.gt_instances.labels = \
                    torch.zeros_like(data_sample.gt_instances.labels)
```
However, we can keep the assignment progress of RPN the same as that in RCNN, and only re-map the learning targets of labels/logits to zero just before the loss calculation.
This unifies and simplifies the code and assignment.
